examples/data_preprocessing/police_killing/preprocessing_rules_from_paper.ipynb

```diff
@@ -0,0 +1,1009 @@
+       "  <thead>\n",
+       "    <tr style=\"text-align: right;\">\n",
+       "      <th></th>\n",
+       "      <th>index</th>\n",
+       "      <th>samples</th>\n",
+       "      <th>rules</th>\n",
+       "      <th>enc_rules</th>\n",
+       "    </tr>\n",
+       "  </thead>\n",
+       "  <tbody>\n",
+       "    <tr>\n",
+       "      <th>0</th>\n",
+       "      <td>0</td>\n",
+       "      <td>Two years earlier , Officer TARGET was killed ...</td>\n",
+       "      <td>[officer killed]</td>\n",
+       "      <td>[24]</td>\n",
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th>1</th>\n",
+       "      <td>1</td>\n",
+       "      <td>Police Chief PERSON said Randolph was found sh...</td>\n",
+       "      <td>[police shot]</td>\n",
+       "      <td>[5]</td>\n",
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th>2</th>\n",
+       "      <td>2</td>\n",
+       "      <td>In the latest incident , Chief Superintendent ...</td>\n",
+       "      <td>[police shots]</td>\n",
+       "      <td>[6]</td>\n",
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th>3</th>\n",
+       "      <td>3</td>\n",
+       "      <td>Chief TARGET of Penn Township police entered t...</td>\n",
+       "      <td>[police shot]</td>\n",
+       "      <td>[5]</td>\n",
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th>4</th>\n",
+       "      <td>4</td>\n",
+       "      <td>A man was was fatally shot by a police officer...</td>\n",
+       "      <td>[police shot, officer shot]</td>\n",
+       "      <td>[5, 25]</td>\n",
+       "    </tr>\n",
+       "  </tbody>\n",
+       "</table>\n",
+       "</div>"
+      ],
+      "text/plain": [
+       "   index                                            samples  \\\n",
+       "0      0  Two years earlier , Officer TARGET was killed ...   \n",
+       "1      1  Police Chief PERSON said Randolph was found sh...   \n",
+       "2      2  In the latest incident , Chief Superintendent ...   \n",
+       "3      3  Chief TARGET of Penn Township police entered t...   \n",
+       "4      4  A man was was fatally shot by a police officer...   \n",
+       "\n",
+       "                         rules enc_rules  \n",
+       "0             [officer killed]      [24]  \n",
+       "1                [police shot]       [5]  \n",
+       "2               [police shots]       [6]  \n",
+       "3                [police shot]       [5]  \n",
+       "4  [police shot, officer shot]   [5, 25]  "
+      ]
+     },
+     "execution_count": 43,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "train_data = get_df(rule2rule_id, df_train)\n",
+    "train_data.head()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "**getting the dev and test data**\n",
+    "\n",
+    "(same as for the train data, moreover the labels and label_ids are added)"
```

Comment:
turn brief comment to the full sentence ("Here we are doing the same as for...")